packages/cozy-harvest-lib/src/datacards/FileDataCard.jsx

```diff
@@ -174,23 +176,58 @@ const makeQueryFromProps = ({ accountId, trigger }) => ({
     .indexFields(['dir_id', 'cozyMetadata.createdAt'])
     .sortBy([{ dir_id: 'desc' }, { 'cozyMetadata.createdAt': 'desc' }])
     .limitBy(5),
-  as: `fileDataCard_io.cozy.accounts/${accountId}/io.cozy.files`,
+  as: `fileDataCard_io.cozy.files/${trigger.message.folder_to_save}/io.cozy.files`,
   fetchPolicy: CozyClient.fetchPolicies.olderThan(30 * 1000)
 })
 
-const FileDataCard = ({ filesCol, konnector, trigger }) => {
-  const { data: files } = filesCol
+const makeSourceAccountQueryFromProps = ({ accountId }) => ({
+  query: Q('io.cozy.files')
+    .where({
+      'cozyMetadata.sourceAccount': accountId,
+      trashed: false
+    })
+    .indexFields(['cozyMetadata.sourceAccount', 'cozyMetadata.createdAt'])
+    .sortBy([
+      { 'cozyMetadata.sourceAccount': 'desc' },
+      { 'cozyMetadata.createdAt': 'desc' }
+    ])
+    .limitBy(5),
+  as: `fileDataCard_io.cozy.accounts/${accountId}/io.cozy.files`,
+  fetchPolicy: CozyClient.fetchPolicies.olderThan(30 * 1000)
+})
 
-  const noFiles = hasQueryBeenLoaded(filesCol) && files.length == 0
-  const isLoading = isQueryLoading(filesCol)
+const FileDataCard = ({
+  folderToSaveFiles,
+  sourceAccountFiles,
+  konnector,
+  trigger
+}) => {
+  const { data: files1 } = folderToSaveFiles
+  const { data: files2 } = sourceAccountFiles
+
+  const noFiles =
+    hasQueryBeenLoaded(folderToSaveFiles) &&
+    files1.length === 0 &&
+    hasQueryBeenLoaded(sourceAccountFiles) &&
+    files2.length === 0
+  const isLoading =
+    isQueryLoading(folderToSaveFiles) || isQueryLoading(sourceAccountFiles)
+
+  const files = useMemo(() => {
+    return sortBy(uniq([...files1, ...files2], x => x._id), x =>
+      get(x, 'cozyMetadata.createdAt')
+    )
+      .reverse()
+      .slice(0, 5)
+  }, [files1, files2])
```

Comment:
Shouldn't we use only files from sourceAccountFiles if we have some?

Reply:
I do not see why we would do that ?

Reply:
Because we can add / move files manually to the `destination folder` and these files will not be imported from the konnector but will be displayed as it it the case?

Reply:
Also, what about if we have 2 accounts for one konnector? If we list by the dest folder, we'll display the same documents twice, no?

Reply:
> Yes, in this case, but then there is not other way to discriminate a file that has been downloaded by the connector if there is no sourceAccount.

To me, if the request on the sourceAccount is returning something, we should not run the query on the destination folder since it's here only for a fallback, no? If there is no result on this sourceAccount query then we should query on the dest folder. 

> Also, what about if we have 2 accounts for one konnector? If we list by the dest folder, we'll display the same documents twice, no?

> I am not sure I understand, the documents are deduplicated by their _id, so we cannot have the same documents twice. If a konnector with multiple accounts arrange their files in the same folder, the files will be shown in the "account 1" list and in the "account 2" list, but as for the previous case, if the konnector does not save the sourceAccount (or has not saved the sourceAccount), we have no way of discriminating.

Let's say we have an OAuth account on the Foo konnector, we don't have much informations to create the destination folder, what will be the path resulting of the creation? `/Administrative/Foo/` ? or `/Administrative/Foo/XXX` where XXX is an ID coming from the OAuth connexion? 

My concern was about the fact we have 2 accounts connected on the same folder `/Administrative/Foo`, in this case, we'll ending up by displaying the same files for account 1 and account 2 if we do not discriminate by the sourceAccount (no dedup involved in this case). So this is the same result as 1, if we can discriminate on sourceAccount we should only use that. If the case is not possible since 2 accounts can't share the same destination folder, let's ignore this case (not sure it's more understandable 🤔 )

Reply:
> To me, if the request on the sourceAccount is returning something, we should not run the query on the destination folder since it's here only for a fallback, no?

If a konnector starts to add a sourceAccount on the data, you do not want the previously saved data to "disappear" from the list right ?

> Let's say we have an OAuth account on the Foo konnector, we don't have much informations to create the destination folder, what will be the path resulting of the creation? /Administrative/Foo/ ? or /Administrative/Foo/XXX where XXX is an ID coming from the OAuth connexion?

I think it depends on the connector, but I think we should have recent konnectors that create the administrative folder once they have enough information.

> My concern was about the fact we have 2 accounts connected on the same folder /Administrative/Foo, in this case, we'll ending up by displaying the same files for account 1 and account 2 if we do not discriminate by the sourceAccount 

Yes, this is true and I personally think it is a better behavior than to only show the files from the administrative folder. We have to keep in mind that as konnector will evolve, at some point the sourceAccounts files will be displayed in priority (thanks to the sort on creation date) over the files from the folder.